plugins/events-backend-module-bitbucket-server/README.md

```diff
@@ -0,0 +1,95 @@
+# events-backend-module-bitbucket-server
+
+Welcome to the `events-backend-module-bitbucket-server` backend plugin!
+
+This plugin is a module for the `events-backend` backend plugin
+and extends it with an `BitbucketServerEventRouter`.
+
+The event router will subscribe to the topic `bitbucketServer`
+and route the events to more concrete topics based on the value
+of the provided `x-event-key` metadata field.
+
+Examples:
+
+| x-event-key         | topic                               |
+| ------------------- | ----------------------------------- |
+| `repo:refs_changed` | `bitbucketServer.repo:refs_changed` |
+| `repo:modified`     | `bitbucketServer.repo:modified`     |
+
+Please find all possible webhook event types at the
+[official documentation](https://confluence.atlassian.com/bitbucketserver/event-payload-938025882.html).
+
+## Installation
+
+# Setting up the router
+
+Install the [`events-backend` plugin](../events-backend/README.md).
+
+Install this module:
+
+```bash
+# From your Backstage root directory
+yarn add --cwd packages/backend @backstage/plugin-events-backend-module-bitbucket-server
+```
+
+Add the event router to the `EventsBackend` instance in `packages/backend/src/plugins/events.ts`:
+
+```diff
++const bitbucketServerEventRouter = new BitbucketServerEventRouter();
+
+new EventsBackend(env.logger)
++  .addPublishers(bitbucketServerEventRouter)
++  .addSubscribers(bitbucketServerEventRouter);
+// [...]
+```
+
+# Setting up the Bitbucket Server Provider
```

Comment:
This section seems a bit unrelated to this plugin itself. The router itself isn't necessarily limited to the provider use case, it doesn't have to be used in combination with it either. Its only purpose is to make events more specific. I understand the intention, though.

Maybe it makes sense to focus on the `BitbucketServerEntityProvider` with events support guide at related `discovery.md` from the related docs? 🤔

Reply:
I don't have a strong preference here. Where I ended up getting confused was the fact that I had to read:
- events-backend
- bitbucket-server-provider
- bitbucket-server-events-backend

Keeping track of all the setup steps across the three guides got me confused during the setup step for the `bitbucketServer` http events topic, which was 2 levels down in links of guides to follow (and was a generic document that didn't have an app-config.yaml example specific for bitbucketServer).

Reply:
Makes sense to provide a holistic view at `discovery.md` then. I believe this is where people would start. And instead of just a reference to the docs for the http topic support, an example for the specifics of its use in that context could be added.

Reply:
I will work on including more references in the `discovery.md`.